x/evm/keeper/abci.go

```diff
@@ -57,5 +64,8 @@ func (k Keeper) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.Valid
 	bloom := ethtypes.BytesToBloom(k.Bloom.Bytes())
 	k.SetBlockBloom(ctx, req.Height, bloom)
 
+	//set the `next revision id` to store
+	k.SetRevisionID(ctx, k.CommitStateDB.GetRevisionID())
```

Comment:
yeah so this should just be a `k.SetRevisionID(ctx)` that sets the csdb revision ID to the store. There's no need for the code on BeginBlock

Reply:
done